### PR TITLE
Ironsides - Tooltips for the bar items

### LINKED ISSUE #3025: Bar data items should have tooltips
### Dev Home version

0.503

### Windows build number

10.0.22631.3672

### Other software

_No response_

### Steps to reproduce the bug

We have several data items (PID, resource usage) on the bar - these should all have tooltips to indicate their meaning. Including the target process name.

### Expected result

_No response_

### Actual result

Missing tooltips 

### Included System Information

_No response_

### Included Extensions Information

_No response_

### PR DESCRIPTION
**Pull request by timkur**
## References and relevant issues
Adds tooltips for the bar items that didn't have them.

## PR checklist
- [ ] Closes #3025
- [ ] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

**Review by krschau — APPROVED** [review]

**Review by jaholme — APPROVED** [review]
:shipit:

**Review by zadesai — APPROVED** [review]

File: tools/PI/DevHome.PI/BarWindowHorizontal.xaml [hunk 1]
```diff
@@ -65,15 +65,15 @@
...
             <TextBlock x:Uid="AppPID" Text="{x:Bind _viewModel.ApplicationPid, Mode=OneWay}" FontFamily="Segoe UI" FontSize="10" Margin="5" VerticalAlignment="Center" Visibility="{x:Bind _viewModel.AppBarVisibility, Mode=OneWay}"/>
             <TextBlock x:Uid="AppCPUUsage" Text="{x:Bind _viewModel.AppCpuUsage, Mode=OneWay}" FontFamily="Segoe UI" FontSize="10" Margin="5" VerticalAlignment="Center" Visibility="{x:Bind _viewModel.AppBarVisibility, Mode=OneWay}"/>
 
             <!-- Per System controls -->
             <Button x:Uid="SwitchLayoutButton" x:Name="SwitchLayoutButton" HorizontalAlignment="Center" Click="{x:Bind _viewModel.SwitchLayoutCommand}">
                 <TextBlock Text="&#xe8b4;"/>
             </Button>
-            <Button x:Name="LargeContentButton" Click="{x:Bind _viewModel.ShowBigWindowCommand}" HorizontalAlignment="Center">
+            <Button x:Name="LargeContentButton" x:Uid="LargeContentButton" Click="{x:Bind _viewModel.ShowBigWindowCommand}" HorizontalAlignment="Center">
```
Comment (dhoehna) [inline tools/PI/DevHome.PI/BarWindowHorizontal.xaml:76]: NIT: a more descriptive Id. "LargeContentButton" is ambiguous.

File: tools/PI/DevHome.PI/BarWindowVertical.xaml [hunk 2]
```diff
@@ -68,15 +70,15 @@
             </controls:ProcessSelectionButton>
 
             <!-- Per App controls -->
-            <Image x:Name="AppIcon" HorizontalAlignment="Center" Source="{x:Bind _viewModel.ApplicationIcon, Mode=OneWay}" Margin="5" Height="20" Width="20" Visibility="{x:Bind _viewModel.AppBarVisibility, Mode=OneWay}"/>
+            <Image x:Name="AppIcon" HorizontalAlignment="Center" Source="{x:Bind _viewModel.ApplicationIcon, Mode=OneWay}" ToolTipService.ToolTip="{x:Bind _viewModel.ApplicationName, Mode=OneWay}" Margin="5" Height="20" Width="20" Visibility="{x:Bind _viewModel.AppBarVisibility, Mode=OneWay}"/>
```
Comment (dhoehna) [inline tools/PI/DevHome.PI/BarWindowVertical.xaml:73]: Please put each property on their own line.

File: tools/PI/DevHome.PI/BarWindowVertical.xaml [hunk 3]
```diff
@@ -68,15 +70,15 @@
...
             <TextBlock x:Uid="AppPID" Text="{x:Bind _viewModel.ApplicationPid, Mode=OneWay}" FontFamily="Segoe UI" FontSize="10" Margin="5" HorizontalAlignment="Center" Visibility="{x:Bind _viewModel.AppBarVisibility, Mode=OneWay}"/>
             <TextBlock x:Uid="AppCPUUsage" Text="{x:Bind _viewModel.AppCpuUsage, Mode=OneWay}" FontFamily="Segoe UI" FontSize="10" Margin="5" HorizontalAlignment="Center" Visibility="{x:Bind _viewModel.AppBarVisibility, Mode=OneWay}"/>
            
             <!-- Per System controls -->
             <Button x:Uid="SwitchLayoutButton" x:Name="SwitchLayoutButton" Click="{x:Bind _viewModel.SwitchLayoutCommand}" HorizontalAlignment="Center">
                 <TextBlock Text="&#xe8b4;"/>
             </Button>
-            <Button x:Name="LargeContentButton" Click="{x:Bind _viewModel.ShowBigWindowCommand}" HorizontalAlignment="Center">
+            <Button x:Name="LargeContentButton" x:Uid="LargeContentButton" Click="{x:Bind _viewModel.ShowBigWindowCommand}" HorizontalAlignment="Center">
```
Comment (dhoehna) [inline tools/PI/DevHome.PI/BarWindowVertical.xaml:81]: Is having duplicate Ids allowed?

File: tools/PI/DevHome.PI/ViewModels/BarWindowViewModel.cs [hunk 4]
```diff
@@ -51,6 +51,9 @@ public partial class BarWindowViewModel : ObservableObject
     [ObservableProperty]
     private Visibility _appBarVisibility = Visibility.Visible;
 
+    [ObservableProperty]
+    private string? _applicationName;
```
Comment (dhoehna) [inline tools/PI/DevHome.PI/ViewModels/BarWindowViewModel.cs:55]: NIT: remove `?` and assign `string.empty` in the constructor.

Comment (timkur) [reply tools/PI/DevHome.PI/ViewModels/BarWindowViewModel.cs:55]: Ok. Seems to fit the pattern better for the rest of the file. I'm guessing a tooltip of null is different than a tooltip of "", but in theory the user won't have access to the tooltip of an app if we're not attached to an app.

File: tools/PI/DevHome.PI/BarWindowHorizontal.xaml (hunk 1 above)
Comment (timkur) [reply tools/PI/DevHome.PI/BarWindowHorizontal.xaml:76]: Good point.

File: tools/PI/DevHome.PI/BarWindowVertical.xaml (hunk 2 above)
Comment (timkur) [reply tools/PI/DevHome.PI/BarWindowVertical.xaml:73]: Just my 2cents... obvious consistency is important, but my vote would be to have all the properties on one line. When I'm trying to see how all of my controls fit into a layout on the page, having a line per control is more readable (to me) than having a single control take up a quarter of the page. I think this borders on personal preference though...

Comment (krschau) [reply tools/PI/DevHome.PI/BarWindowVertical.xaml:73]: Just to add another two cents🙂... generally, we put everything on one line when it's short enough to fit and split when it's not. Personally, I find that easier to read in an IDE, but I also think splitting long lines makes it easier to review on GitHub. I'm fine with the wontfix, just wanted to add a little context.

**Pull request merged**